core/bootstrap/src/main/java/org/wildfly/swarm/bootstrap/modules/ApplicationModuleFinder.java

```diff
@@ -121,7 +120,7 @@ protected void addAsset(ModuleSpec.Builder builder, ApplicationEnvironment env)
             name = name.substring(0, dotLoc);
         }
 
-        File tmp = TempFileManager.INSTANCE.newTempFile(name, ext);
+        File tmp = File.createTempFile(name, ext);
```

Comment:
Good question @Ladicek. It's needed in this case as we need to close the jar/war loader first before deleting the file. And, as shutdown hooks are executed in a non deterministic order, we can't ensure when it would be.